### PR TITLE
validate-and-update-endpoint-workflow

### PR DESCRIPTION
**Pull request by Nil20**

### CODE REVIEW

File: packages/client/src/transformer/index.ts [hunk 1]
```diff
@@ -134,43 +122,43 @@ const transformRegistrationCorrection = (
...
-    values.push(payload)
+    })
   }
-
-  transformedData.registration.correction.values = values
+  return changedValues
 }
 
-export function addCorrectionDetails(
+export function getChangedValues(
```
Comment (rikukissa) [inline packages/client/src/transformer/index.ts:154]: Very good improvement 👍

**Review by rikukissa — CHANGES_REQUESTED** [review]
Looking great! One thing I was expecting to find are the removal of the branches of the old event detection. Is that still in the works @Nil20 ?

File: packages/commons/src/fhir/index.ts [hunk 2]
```diff
@@ -74,6 +74,19 @@ export type WithUUID<T extends Resource> = Omit<T, 'id'> & {
...
+  | 'CORRECTION_REQUESTED'
+  | 'DECLARATION_UPDATED'
+  | 'DECLARED'
+  | 'IN_PROGRESS'
+  | 'ISSUED'
+  | 'REGISTERED'
+  | 'REJECTED'
+  | 'VALIDATED'
+  | 'WAITING_VALIDATION'
+
```
[the remark this answers is not in the document]
Comment (Nil20) [reply packages/commons/src/fhir/index.ts:89]: Yeah, correct! These don't have anything to do with fhir really. We can move them in the `record.ts`

File: packages/workflow/src/records/fhir.ts [hunk 3]
```diff
@@ -299,6 +299,110 @@ export function createCorrectedTask(
...
+    requester: {
+      agent: { reference: `Practitioner/${practitioner.id}` }
+    },
+    identifier: previousTask.identifier,
+    extension: [
+      ...previousTask.extension.filter((extension) =>
+        [
+          'http://opencrvs.org/specs/extension/contact-person-phone-number',
+          'http://opencrvs.org/specs/extension/informants-signature',
+          'http://opencrvs.org/specs/extension/contact-person-email'
```
Comment (Zangetsu101) [inline packages/workflow/src/records/fhir.ts:323]: There should be a similar extension for mobile number

File: packages/workflow/src/records/fhir.ts [hunk 4]
```diff
@@ -299,6 +299,110 @@ export function createCorrectedTask(
...
+        ].includes(extension.url)
+      ),
+      {
+        url: 'http://opencrvs.org/specs/extension/timeLoggedMS',
+        valueInteger: 0
+      }
```
Comment (Zangetsu101) [inline packages/workflow/src/records/fhir.ts:329]: This value should come from the client payload

File: packages/workflow/src/records/state-transitions.ts [hunk 5]
```diff
@@ -184,6 +189,80 @@ export async function toCorrectionApproved(
...
+    ...record.entry.map((entry) => {
+      const { fullUrl, ...restOfResources } = entry
+
+      if (restOfResources.resource.id !== previousTask.id) {
+        return entry
+      }
+      return {
+        fullUrl,
+        resource: updatedTaskWithLocationExtensions
+      }
```
Comment (Zangetsu101) [inline packages/workflow/src/records/state-transitions.ts:224]:
```suggestion
      if (entry.resource.id !== previousTask.id) {
        return entry
      }
      return {
        ...entry,
        resource: updatedTaskWithLocationExtensions
      }
```

File: packages/workflow/src/records/state-transitions.ts [hunk 6]
```diff
@@ -281,7 +360,7 @@ export async function toCorrectionRejected(
     practitioner
   )
 
-  const taskHistory = await getTaskHistory(currentCorrectionRequestedTask.id)
+  const taskHistory = await getTaskHistory(currentCorrectionRequestedTask.id!)
```
Comment (Zangetsu101) [inline packages/workflow/src/records/state-transitions.ts:363]: This shouldn't be needed

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

**Comment by rikukissa** [comment]
> > Looking great! One thing I was expecting to find are the removal of the branches of the old event detection. Is that still in the works @Nil20 ?
> 
> @rikukissa I think it would be better to remove them at the end when all the endpoints are done no?

Both are fine as long as they get removed 👍

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**